### ISSUE TITLE
Provide an API for long running transactions with dependencies between write-operations

### ISSUE THREAD

**Issue by nikolasburk** [prisma/prisma#3846]
There already is a [GitHub issue](https://github.com/prisma/prisma/issues/74) asking for a way to submit multiple mutations within the same HTTP request where all mutations are executed as one transaction.

However, this feature does not allow for creating a transaction where the mutations depend on each other. Here is an example where the second operation depends on the first:

```js
prisma.transaction(async tx => {
  const user = await tx.createUser({
    name,
  })
  // compute some stuff
  await tx.updateOrder(user.id)
  await tx.commit()
})
```

This is currently not possible with the Prisma API as it would require having a long-running connection where the results of the operations are sent back and forth between Prisma and the database. 

It should be considered whether it is helpful to add such an API to Prisma, or whether abstractions like nested mutations and the [requested transactional execution of multiple _independent_ transactions](https://github.com/prisma/prisma/issues/74) are the better approach for Prisma.

**Comment by mattwilson1024** [comment]
👍 This seems like a very useful (if not essential) feature to me and has been the only real pain point for me so far as a new Prisma user.

As you've identified, there are situations where you need to perform several mutations in sequence, making use of the response data as you go, but with the confidence that it'll be a single transaction that rolls back in the case of a failure. In my opinion, neither nested mutations or transactions for independent mutations seem quite sufficient for these use-cases.

If it helps to have example use-cases for discussion, there's a little more detail about my specific requirement here: https://www.prisma.io/forum/t/transactions-when-the-business-logic-requires-more-than-just-a-nested-insert/5904

**Comment by matthewhuang-camelot** [comment]
Is there an ability to write raw SQL via the Prisma API or some other way around this for the meantime? I love what I'm seeing with Prisma, but it seems that this makes it unusable for building more resilient applications at the moment?

**Comment by pantharshit00** [comment]
@matthewhuang-camelot

Yes, raw sql queries are possible see: https://www.prisma.io/docs/prisma-graphql-api/reference/raw-database-access-qwe4/

**Comment by matthewhuang-camelot** [comment]
@pantharshit00 Thanks! Just saw that too :D

**Comment by savioserra** [comment]
Hi guys, any updates on this? :(

**Comment by pantharshit00** [comment]
@savioserra 

We are working on a new client which will have this.

Please check here: https://github.com/prisma/rfcs/blob/new-ts-client-rfc/text/0000-new-ts-client.md

**Comment by savioserra** [comment]
@pantharshit00 Nice :) Gonna take a look. For now, I'm using stored procedures and Prisma's `executeRaw` mutation as a workaround.

**Comment by waleCloud** [comment]
Hello People, any update on this yet? 😕

**Comment by arvinlorenz** [comment]
hi any update on this?

**Comment by iamclaytonray** [comment]
Will this be part of Prisma 2?

**Comment by Ilyes512** [comment]
Looking into ORM's for Node, Prism looks REALLY good (website, documentation, activity on Github, etc). But lacking/missing (manual) migrations is a HUGE no-go for me. I looked at the Prisma 2 change-log and article and I only something see nested migrations, which is not the same (it can't replace manual transactions (grouping)).
